### PR TITLE
Used relative stream to simplify emitBodyRange. Removes necessity to check lower bound

### PR DESCRIPTION
**Pull request by hannesvdvreken**
Replaces #188

### CODE REVIEW

File: src/Response/SapiStreamEmitter.php [hunk 1]
```diff
@@ -78,21 +79,14 @@ private function emitBodyRange(array $range, ResponseInterface $response, $maxBu
...
-        if (!$body->isSeekable()) {
-            $contents = $body->getContents();
-            echo substr($contents, $first, $last - $first);
-            return;
-        }
-
-        $body->seek($first);
-        $pos = $first;
+        $body = new RelativeStream($response->getBody(), $first);
+        $body->rewind();
```
Comment (weierophinney) [inline src/Response/SapiStreamEmitter.php]: This directly conflicts with #200 (represented in lines 84 - 88 above), which provided a fix that ensures that the emitter can be used with non-seekable streams. Please reinstate that block.

Comment (hannesvdvreken) [inline src/Response/SapiStreamEmitter.php]: Yes, I saw that. Will find some time to fix that asap.

**Comment by hannesvdvreken** [comment]
@weierophinney should be fixed now 😉 waiting for the build.

**Comment by weierophinney** [comment]
Looks like the rebase dropped the lines entirely around testing for seekable streams, @hannesvdvreken — please see: https://travis-ci.org/zendframework/zend-diactoros/jobs/158205643#L240-L248

**Comment by weierophinney** [comment]
@hannesvdvreken I can see how to fix this; I'll take it from here.

**Comment by weierophinney** [comment]
Thanks, @hannesvdvreken

**Comment by hannesvdvreken** [comment]
I was away from computer for a while. Thanks for finishing this, @weierophinney.

**Comment by weierophinney** [comment]
My pleasure, @hannesvdvreken — thanks for the patch!

**Pull request merged**